### ISSUE TITLE
Ylims of data that is below 10^0 are incorrectly inferred.

### ISSUE THREAD

**Issue by vchuravy** [MakieOrg/Makie.jl#4513]
Using CairoMakie & Makie v0.12.13

```julia
f(x, a=0.3, b=3.0; N=10) = sum(a^k * cos(b^k * π * x) for k in 0:N)
f′(x, a=0.3, b=3.0; N=10) = - sum(a^k * b^k * π * sin(b^k * π * x) for k in 0:N)

function csfd(F, u, v, ϵ)
	imag.(F(u .+ v.* ϵ*im)) ./ ϵ
end

function rel_error(f, f′, x, ϵ)
	y′ = f′(x)
	y = csfd(f, x, one(x), ϵ)
	abs((y′-y)/y′)
end

ϵs = [10.0^-x for x in 30:-0.1:4]
x = 0.2
re = map(ϵ->rel_error((x)->f(x, N=3), (x)->f′(x, N=3), x, ϵ), ϵs)
```

The initial plot looks fine:

```
lines(ϵs,re)
```

![Image](https://github.com/user-attachments/assets/5e1cf2af-746c-4db6-9539-777aba1c934a)

But since `ϵs` is exponential plotting on a log scale makes more sense.

```
lines(ϵs, re, axis=(;xscale=log10, yscale=log10))
```

![Image](https://github.com/user-attachments/assets/b753ae0b-2ec1-4b4c-be55-ca0354308605)

Manually specifying the axis.

```
fig, ax, _ = lines(ϵs, re, axis=(;xscale=log10, yscale=log10))
ylims!(ax, 1e-20, 10e1)
fig
```

![Image](https://github.com/user-attachments/assets/7424ac85-facc-4490-a3a5-f87868d4709c)

Ideally  the ylims would be automatically detected to be going to `1e-20` and I can't quite explain the odd "gaps".

The plot I was trying to reproduce is:

![Image](https://github.com/user-attachments/assets/40036ce9-1fba-4405-adce-dedb20a18579)

from https://www.sciencedirect.com/science/article/pii/S037704272100354X?via%3Dihub#fig3

and currently it looks like:

![Image](https://github.com/user-attachments/assets/2192fd19-24ec-4958-b15f-86fea687cb85)

**Comment by EdsterG** [comment]
The odd gaps are probably due to NaN or Inf at those points.

**Comment by vchuravy** [comment]
Yeah likely since the original data is 0 and then the yscale turns it into NaN.

Would be nice if the axis scale would be robust to that

**Comment by asinghvi17** [comment]
The Axis implementation defaults to `data_limits` if there is no transform_func, and `boundingbox` otherwise.

```julia

julia> Makie.data_limits(ax.scene)
GeometryBasics.HyperRectangle{3, Float64}([1.0e-30, 0.0, 0.0], [0.0001, 3.212426380795015e-5, 0.0])

julia> ax.scene.plots
1-element Vector{Plot}:
 Lines{Tuple{Vector{Point{2, Float64}}}}

julia> Makie.boundingbox(ax.scene)
GeometryBasics.HyperRectangle{3, Float64}([-30.0, Inf, 0.0], [26.0, NaN, 0.0])

```
The problem here is that `boundingbox` applies the transform to the boundingbox from `data_limits`, meaning it doesn't have access to the individual coordinates so it can filter them out.

https://github.com/MakieOrg/Makie.jl/blob/a1c513b73f6238f6fcbfcc3d7c9d5ba41493e2a7/src/layouting/boundingbox.jl#L35-L40

We could have `boundingbox` iterate through the coordinates of the plot manually, in the same way `data_limits` does, and compute the boundingbox from the transformed points, not the boundingbox in data space.  I think it used to do this at some point and then we changed that behaviour @ffreyer?

**Comment by vchuravy** [comment]
> floating-point error.

Almost, but not quite. The goal of the program is to calculate the relative error (in floating point) between two different ways of calculating the derivative of a program. 

`f'` is the analytical derivative and the other way uses a complex-step finite difference method. Expressing the relative error here is dependent on IEEE Float so going to BigFloat calculates something different.

So getting the relative error to be `0` is actually correct for me. Plotting that on a log-scale is of course weird...

Now I don't know how the original figure was created, they did use a lot fewer sample points, maybe they choose those points where the error didn't occur? 

In any case the main issue for me is the `ylims`

**Comment by vchuravy** [comment]
@asinghvi17 re BigFloat see 

![Image](https://github.com/user-attachments/assets/4b73464f-eb77-4dcf-8bca-059000d887a8)

Looks kinda nice, but falsifies the error in Float64.

**Comment by asinghvi17** [comment]
Thanks, that makes more sense. 

Yeah the ylim bit can be solved well enough via a clever definition of `boundingbox`. We just need to change that definition again, which would also be beneficial for GeoMakie but I'm interested to see Frederic's take on this since he's working pretty closely with that code.

**Comment by asinghvi17** [comment]
I was going to say earlier but lost the comment - for now, you could `clamp` to an appropriately low value.  Something like `clamp.(re, 1e-30, Inf)` or so.   That should solve the plotting errors for the moment.

**Comment by ffreyer** [comment]
> We could have `boundingbox` iterate through the coordinates of the plot manually, in the same way `data_limits` does, and compute the boundingbox from the transformed points, not the boundingbox in data space. I think it used to do this at some point and then we changed that behaviour [@ffreyer](https://github.com/ffreyer)?

It probably looked like a useless performance drain to me. Applying a function + doing a matmult on one Rect is a lot cheaper than potentially a million points or so.

**Comment by asinghvi17** [comment]
This is true...maybe we need transformation traits to say whether doing that is safe or unsafe?
